### PR TITLE
Updated registry references to registry.k8s.io

### PR DESCRIPTION
**Pull request by rjsadow**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
ref:  https://github.com/kubernetes/release/issues/2904
For https://github.com/kubernetes/k8s.io/issues/4738

#### Which issue(s) this PR fixes:
None, partial completion of https://github.com/kubernetes/release/issues/2904

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: images/build/debian-iptables/README.md [hunk 1]
```diff
@@ -11,11 +11,11 @@ If you're editing the Dockerfile or some other thing, please bump the `TAG` in t
...
-# ---> staging-k8s.gcr.io/debian-iptables-amd64:TAG
-# ---> staging-k8s.gcr.io/debian-iptables-arm:TAG
-# ---> staging-k8s.gcr.io/debian-iptables-arm64:TAG
-# ---> staging-k8s.gcr.io/debian-iptables-ppc64le:TAG
-# ---> staging-k8s.gcr.io/debian-iptables-s390x:TAG
+# ---> staging-registry.k8s.io/debian-iptables-amd64:TAG
+# ---> staging-registry.k8s.io/debian-iptables-arm:TAG
+# ---> staging-registry.k8s.io/debian-iptables-arm64:TAG
+# ---> staging-registry.k8s.io/debian-iptables-ppc64le:TAG
+# ---> staging-registry.k8s.io/debian-iptables-s390x:TAG
```
Comment (sftim) [inline images/build/debian-iptables/README.md:18]: I'm not sure these are right

File: images/build/distroless-iptables/README.md [hunk 2]
```diff
@@ -11,11 +11,11 @@ If you're editing the Dockerfile or some other thing, please bump the `TAG` in t
...
-# ---> staging-k8s.gcr.io/distroless-iptables-amd64:TAG
-# ---> staging-k8s.gcr.io/distroless-iptables-arm:TAG
-# ---> staging-k8s.gcr.io/distroless-iptables-arm64:TAG
-# ---> staging-k8s.gcr.io/distroless-iptables-ppc64le:TAG
-# ---> staging-k8s.gcr.io/distroless-iptables-s390x:TAG
+# ---> staging-registry.k8s.io/distroless-iptables-amd64:TAG
+# ---> staging-registry.k8s.io/distroless-iptables-arm:TAG
+# ---> staging-registry.k8s.io/distroless-iptables-arm64:TAG
+# ---> staging-registry.k8s.io/distroless-iptables-ppc64le:TAG
+# ---> staging-registry.k8s.io/distroless-iptables-s390x:TAG
```
Comment (sftim) [inline images/build/distroless-iptables/README.md:18]: I'm not sure these are right

File: images/build/setcap/README.md [hunk 3]
```diff
@@ -15,11 +15,11 @@ Build and push images for all the architectures:
...
-# ---> staging-k8s.gcr.io/setcap-amd64:TAG
-# ---> staging-k8s.gcr.io/setcap-arm:TAG
-# ---> staging-k8s.gcr.io/setcap-arm64:TAG
-# ---> staging-k8s.gcr.io/setcap-ppc64le:TAG
-# ---> staging-k8s.gcr.io/setcap-s390x:TAG
+# ---> staging-registry.k8s.io/setcap-amd64:TAG
+# ---> staging-registry.k8s.io/setcap-arm:TAG
+# ---> staging-registry.k8s.io/setcap-arm64:TAG
+# ---> staging-registry.k8s.io/setcap-ppc64le:TAG
+# ---> staging-registry.k8s.io/setcap-s390x:TAG
```
Comment (sftim) [inline images/build/setcap/README.md:22]: I'm not sure these are right

File: images/build/debian-iptables/README.md (hunk 1 above)
Comment (rjsadow) [reply images/build/debian-iptables/README.md:18]: I think you're right, but I also don't think it's correct as it currently is. When I run the make file it seems to want to push to `gcr.io/k8s-staging-build-image/debian-iptables-${ARCH}:${TAG}` I'll update the file with the correct reference.

**Review by sftim — COMMENTED** [review]
/lgtm

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**